pkg/flow/flow.go

```diff
@@ -155,6 +160,11 @@ func newController(o controllerOptions) *Flow {
 		}
 	}
 
+	if workerPool == nil {
+		level.Info(log).Log("msg", "no worker pool provided, creating a default pool")
+		workerPool = worker.NewDefaultWorkerPool()
```

Comment:
I wonder if we should provide a way for users to configure this worker pool? Maybe if they run very agents with loads of components they'd a want larger pool size?

Reply:
It is possible. Previously, however, we were running with one thread doing all this work, so I don't think this will be something we hit soon. I would be open to making it configurable in the future, but we may not need it for a while.